include/verilated_vpi.cpp

```diff
@@ -1749,6 +1749,11 @@ bool vl_check_format(const VerilatedVar* varp, const p_vpi_value valuep, const c
         case VLVT_UINT32: return status;
         default: status = false;
         }
+    } else if (valuep->format == vpiRealVal) {
+        switch (varp->vltype()) {
+        case VLVT_UINT64: return status;
```

Comment:
```suggestion
        case VLVT_REAL: return status;
```

Then you need plumb backwards to add VLVT_REAL, which will reach into the Verilator sources.